### PR TITLE
Start prompt app | Prompts [1/X]

### PR DESCRIPTION
**Pull request by joycrwu**
## Overview

Started a new Django app for prompts. 

## Changes Made

Filled out a couple of views -- including getting all prompts, creating a prompt, getting a specific prompt, updating a specific prompt, deleting a specific prompt. 

-- Getting all prompts
-- Creating a prompt

## Test Coverage
**Creating a prompt:**
Manual testing via Postman, endpoint: http://127.0.0.1:8000/api/prompts/ POST
Example Request Body: `{
    "question_name" : "Favorite food",
    "question_placeholder" : "Ramen"
}`

Example Response: `{
    "id": 2,
    "question_name": "Favorite food",
    "question_placeholder": "Ramen"
}`

**Getting all prompts**
Manual testing via Postman, endpoint: http://127.0.0.1:8000/api/prompts/ GET
Example Response: `[
    {
        "id": 1,
        "question_name": "WHAT DA FUCK",
        "question_placeholder": "FUCK OFF"
    },
    {
        "id": 2,
        "question_name": "Favorite food",
        "question_placeholder": "Ramen"
    }
]`

## Related PRs and Issues
#23 

## Next Steps (delete if not applicable)

Write out the controllers for the prompts nd update the prompts to reflect what kind of information Django is actually seeking from its users.

### CODE REVIEW

**Review by katesliang — CHANGES_REQUESTED** [review]
good start to prompts! code is slay 

just a few nits: 
- could you create an issue (like #20) that summarizes what you plan to do with this stack?
- could you summarize the new views that you created in your PR (like the endpoints, sample request bodies and responses)?
- it may be better to delete the views that have the controllers for now and then add it in later as we create it so we don't have any unresolved imports

**Review by chalo2000 — APPROVED** [review]
+1 on all of Kate's good points, create issues to describe what you're going to do, have better PR descriptions, and push code that compiles! after addressing her comments you can merge this in.

**Comment by joycrwu** [comment]
> good start to prompts! code is slay
> 
> just a few nits:
> 
> * could you create an issue (like #20) that summarizes what you plan to do with this stack?
> * could you summarize the new views that you created in your PR (like the endpoints, sample request bodies and responses)?
> * it may be better to delete the views that have the controllers for now and then add it in later as we create it so we don't have any unresolved imports

**Comment by joycrwu** [comment]
@katesliang changes made! also made an issue and committed code that the views can reference in the controller now

File: src/prompts/views.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+from api import settings as api_settings
+from api.utils import failure_response
+from api.utils import success_response
+from prompts.models import Prompt
+from rest_framework import generics
+from rest_framework import status
+
+from .controllers.create_prompt_controller import CreatePromptController
+# from .controllers.update_prompt_controller import UpdatePromptController
```
Comment (katesliang) [inline src/prompts/views.py]: delete comments

File: src/api/urls.py [hunk 2]
```diff
@@ -15,6 +16,7 @@
     path("me/", MeView.as_view(), name="me"),
     path("rides/<int:id>/", RideView.as_view(), name="ride"),
     path("rides/", RidesView.as_view(), name="rides"),
-    path("search/", SearchView.as_view(), name="search")
+    path("search/", SearchView.as_view(), name="search"),
     re_path(r"^requests/", include("request.urls")),
+    path("prompts/", PromptsView.as_view(), name="prompts")
```
Comment (katesliang) [inline src/api/urls.py]: nit: could you create separate urls.py for prompts?

File: src/prompts/views.py (hunk 1 above)
Comment (katesliang) [reply src/prompts/views.py]: remember to reinstall the pre commit hooks on your new laptop so that if formats everything nicely for you before committing !

File: src/api/urls.py (hunk 2 above)
Comment (katesliang) [reply src/api/urls.py]: similar to what is done in #17

**Review by katesliang — APPROVED** [review]
nice changes!! pls address my comments before merging~~ 

also when you merge, remember to do "squash and merge" so that the commit history in master is cleaner

pr looks super profesh now

File: src/prompts/controllers/create_prompt_controller.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    def __init__(self, data, serializer):
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        name = self._data.get("question_name")
+        if name is None:
+            return failure_response("No name provided.", status.HTTP_400_BAD_REQUEST)
+        
+        label = self._data.get("question_placeholder")
```
Comment (chalo2000) [inline src/prompts/controllers/create_prompt_controller.py]: why label and not placeholder?

File: src/prompts/views.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    def post(self, request):
+        """Create a prompt."""
+        try:
+            data = json.loads(request.body)
+        except json.JSONDecodeError:
+            data = request.data
+        return CreatePromptController(data, self.serializer_class).process()
+
+
+# class PromptView(generics.GenericAPIView):
```
Comment (chalo2000) [inline src/prompts/views.py]: delete comments

**Review by chalo2000 — APPROVED** [review]
LGTM
